tools/actions_local_runner.py

```diff
@@ -70,65 +78,169 @@ def find_changed_files() -> List[str]:
+        cmd,
+        shell=True,
+        cwd=REPO_ROOT,
+        env=env,
+        stdout=subprocess.PIPE if redirect else None,
+        stderr=subprocess.PIPE if redirect else None,
+        executable=shutil.which("bash"),
+    )
+    stdout, stderr = await proc.communicate()
+
+    passed = proc.returncode == 0
+    if not redirect:
+        return passed, "", ""
+
+    return passed, stdout.decode(), stderr.decode()
+
+
+def header(name: str, passed: bool) -> None:
+    PASS = color(col.GREEN, "\N{check mark}")
+    FAIL = color(col.RED, "x")
+    icon = PASS if passed else FAIL
+    print(f"{icon} {color(col.BLUE, name)}")
+
+
+def get_flake_excludes() -> List[str]:
+    config = configparser.ConfigParser()
+    config.read(os.path.join(REPO_ROOT, ".flake8"))
+
+    excludes = config["flake8"]["exclude"].split("\n")
+    return [i.strip() for i in excludes if i.strip() != ""]
+
+
+async def run_flake8(files: Optional[List[str]], quiet: bool) -> bool:
+    cmd = "flake8"
+    if files is not None:
+        excludes = get_flake_excludes()
+        files = [
+            f
+            for f in files
+            if not any(is_match(pattern=exclude, filename=f) for exclude in excludes)
+        ]
+        if len(files) == 0:
+            print_results("flake8", True, [])
+            return True
+
+        # shlex.join is Python 3.8+, but we guard for that before getting here
+        # so ignore mypy
+        cmd = f"flake8 {shlex.join(files)}"  # type: ignore[attr-defined]
+
+    passed, stdout, stderr = await shell_cmd(cmd)
+    print_results("flake8", passed, [stdout, stderr])
+    return passed
+
+
+async def run_mypy(files: Optional[List[str]], quiet: bool) -> bool:
+    if files is not None:
+        # Running quick lint, use mypy-wrapper instead so it checks that the files
+        # actually should be linted
+        stdout = ""
+        stderr = ""
+        passed = True
+
+        # Pass each file to the mypy_wrapper script
+        # TODO: Fix mypy wrapper to mock mypy's args and take in N files instead
+        # of just 1 at a time
+        for f in files:
+            f = os.path.join(REPO_ROOT, f)
+            f_passed, f_stdout, f_stderr = await shell_cmd(
+                f"{sys.executable} tools/mypy_wrapper.py '{f}'"
+            )
+            if not f_passed:
+                passed = False
+            stdout += f_stdout + "\n"
+            stderr += f_stderr + "\n"
+
+        print_results("mypy (skipped typestub generation)", passed, [stdout, stderr])
+        return passed
+
+    # Not running quicklint, so use lint.yml
+    _, _, _ = await shell_cmd(
+        f"{sys.executable} tools/actions_local_runner.py --job 'mypy' --file .github/workflows/lint.yml --step 'Run autogen'",
+        redirect=False,
+    )
+    passed, _, _ = await shell_cmd(
+        f"{sys.executable} tools/actions_local_runner.py --job 'mypy' --file .github/workflows/lint.yml --step 'Run mypy'",
+        redirect=False,
+    )
```

Comment:
same comment for `shlex.join` with `sys.executable` probably applies here too, but it's up to you whether you want to use it or not